flake.nix

```diff
@@ -1,140 +1,40 @@
-# The flake.nix is the entrypoint of all nix code.
-#
-# This repository uses the standard tool https://github.com/divnix/std.
-# Familiarity with std is required to be able to contribute effectively.
-# While official documentation for std can be found in its GitHub, this flake
-# has been thoroughly commented so as to quickstart new maintainers.
-# This flake can also be used as a template for new std-based projects.
-# Further documentation can be found in nix/README.md
-#
-# You may want to refer to the standard glossary as you go along:
-# https://divnix.github.io/std/glossary.html
 {
   description = "Plutus Core";
 
+
   inputs = {
-    nixpkgs = {
-      follows = "haskell-nix/nixpkgs";
-    };
-    std = {
-      url = "github:divnix/std";
-      inputs.nixpkgs.follows = "nixpkgs";
-    };
-    haskell-nix = {
-      url = "github:input-output-hk/haskell.nix";
-      inputs = {
-        hackage.follows = "hackage-nix";
-      };
+    iogx = {
+      url = "github:input-output-hk/iogx?ref=v4";
+      inputs.hackage.follows = "hackage";
+      inputs.CHaP.follows = "CHaP";
+      inputs.haskell-nix.follows = "haskell-nix";
+      inputs.nixpkgs.follows = "haskell-nix/nixpkgs-2305";
```

Comment:
this changes our nixpkgs pin, not sure if we want that. I guess it's fine

Reply:
These are parts of `iogx` inputs:

```nix
    haskell-nix = {
      url = "github:input-output-hk/haskell.nix/445086f35d450af485cd9200b86211355095b2a1";
      inputs.hackage.follows = "hackage";
    };

    nixpkgs.follows = "haskell-nix/nixpkgs-2305";

    hackage = {
      url = "github:input-output-hk/hackage.nix";
      flake = false;
    };
```

If one uses iogx one must use the `nixpkgs` provided by it, since it will have the necessary overlays, etc.. 

When pinning `haskell-nix` in your flake, nix complained and so I had to add the line `inputs.nixpkgs.follows = "haskell-nix/nixpkgs-2305";`

I guess if one must use another `nixpkgs` one can always add say `nixpkgs-2` to their inputs.

Reply:
> If one uses iogx one must use the nixpkgs provided by it, since it will have the necessary overlays, etc..

I don't see how this affects the choice of flake input?

My issue is just that we were previously using `haskell-nix/nixpkgs` and you changed it to `haskell-nix/nixpkgs-2305`.

Reply:
Got it, will change to `haskell-nix/nixpkgs`